Automattic/jetpack
# Social: Improve broken connection notices

Currently in the editor sidebar, we show a general message that some connections are broken without mentioning which ones and also in the connections management modal, we allow users to connect more accounts when existing ones are broken

## Proposed changes:

* Display list of broken connections in editor sidebar instead of a generic message
* Change the "Broken connections" text in the connections modal to a red color alert
* Replace "Connect more" button with "Fix connections" when there are broken connections
* After this change, users won't be able to add new connections for a service if it has a broken connection

### Other information:


## Jetpack product discussion
p1HpG7-tyu-p2

## Does this pull request change what data or activity we track or use?



## Testing instructions:






<details>
    <summary>To simulate a broken connection, you can make these changes (click to expand)</summary>

```diff
diff --git a/projects/packages/publicize/src/class-publicize.php b/projects/packages/publicize/src/class-publicize.php
index 715d6cd3f8..63e1ee048e 100644
--- a/projects/packages/publicize/src/class-publicize.php
+++ b/projects/packages/publicize/src/class-publicize.php
@@ -255,7 +255,7 @@ class Publicize extends Publicize_Base {
                                                                        'can_disconnect' => self::can_manage_connection( $connection['connection_data'] ),
                                                                        'profile_link'   => $this->get_profile_link( $service_name, $connection ),
                                                                        'shared'         => '0' === $connection['connection_data']['user_id'],
-                                                                       'status'         => 'ok',
+                                                                       'status'         => 'broken',
                                                                )
                                                        );
                                                } else {
@@ -287,7 +287,7 @@ class Publicize extends Publicize_Base {
                $connection_results_map = array();

                foreach ( $connection_results as $connection_result ) {
-                       $connection_results_map[ $connection_result['connection_id'] ] = $connection_result['test_success'] ? 'ok' : 'broken';
+                       $connection_results_map[ $connection_result['connection_id'] ] = $connection_result['test_success'] ? 'broken' : 'broken';
                }
                foreach ( $connections as $key => $connection ) {
                        if ( isset( $connection_results_map[ $connection['connection_id'] ] ) ) {
```
</details>

- Goto post editor
- Confirm that you see a list of connections in the broken connections notice
- Open the connections management modal
- Confirm that you no longer see the yellow color "Broken connections" notice
- Confirm that "Connect more" button is replaced with "Fix connections"
- Click on it
- Confirm that it opens the service panel
- Confirm that Reconnecting fixes the issue (After you remove the simulation code)
- Confirm that it works as expected in the editor for non-admin authors.

<table>
    <tbody>
        <tr>
            <th colspan="2">Editor sidebar</th>
        </tr>
        <tr>
            <th>BEFORE</th>
            <th>AFTER</th>
        </tr>
        <tr>
            <td>
<img width="291" alt="Screenshot 2024-07-22 at 2 41 57 PM" src="https://github.com/user-attachments/assets/06ad3b2c-05fe-4e6e-a62e-ac0cb96a3652">
</td>
            <td>

<img width="275" alt="Screenshot 2024-07-22 at 3 18 59 PM" src="https://github.com/user-attachments/assets/8a01b378-a5b8-490a-890f-8dda9cdee884">
</td>
        </tr>
        <tr>
            <th colspan="2">Connections modal</th>
        </tr>
        <tr>
            <td colspan="2">

https://github.com/user-attachments/assets/7922b585-993f-4fce-b604-c47276a6f438


</td>
        </tr>
    </tbody>
</table>
